src/sysutil.rs

```diff
@@ -191,12 +200,56 @@ impl SysUtil {
                     panic!("Session file '{}' is not UTF-8: {}", path.display(), err)
                 });
 
+                let fname_and_type = match path.strip_prefix(sess_parent_dir) {
+                    Ok(fname_and_type) => fname_and_type.as_os_str().to_os_string(),
```

Comment:
This is done in order to not get `E0597` error regarding borrowing.

Reply:
You can simplify this with `fname_and_type.to_owned()`

Reply:
Awesome, another neat little thing in Rust!